cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/BaseController.java

```diff
@@ -93,10 +127,39 @@ else if (ex instanceof UnsupportedOperationException) {
             // no need for returning different exit codes for different exceptions, but that's also possible in the
             // future
             return 1;
+        });
+
+        // parse args now to be able to configure logging before we continue running the rest of the CLI
+        // a nice catch of this approach: we can properly handle all sorts of argument parsing errors nicely
+        try {
+            cli.parseArgs(args);
+        } catch (CommandLine.PicocliException e) {
+            // TODO: consider printing this directly to stderr
```

Comment:
I think we should not use the logger in this special case, as it produces a more or less "unexpected" output:

```
Jun 16, 2020 10:26:00 PM cloud.foundry.cli.crosscutting.logging.Log log
SEVERE: Error: --verbose, --debug are mutually exclusive (specify only one)
```

Most other CLI tools don't use loggers to log CLI errors, so IMO in this special case we could use `System.err` directly.

Feedback welcome!

Reply:
I think this is fine, 
we can ask the other devs about their opinion.